tests/integration/mount_propagation.bats

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env bats
+
+load helpers
+
+function setup() {
+  teardown_busybox
+  setup_busybox
+
+  # Create a "shared" parent mount point for container
+  # rootfs and make sure pivot_root() still works. This
+  cd ${BUSYBOX_BUNDLE}
+  mount --bind . .
```

Comment:
If you put this inside `@test` and you use `unshare -m` to run a shell command, that might work a bit nicer since it means we won't be messing with host mountpoints while testing.